### PR TITLE
Update to Quarkus 3.8

### PR DESCRIPTION
**Pull request by sophokles73**
Also adapted some unit tests to reflect changes in Kafka and k8s client libraries.

### CODE REVIEW

**Comment by sophokles73** [comment]
@calohmn with the update to Quarkus 3.8. the `org.eclipse.hono.client.kafka.consumer.AsyncHandlingAutoCommitKafkaConsumerTest` has started to fail. My assumption is that the behavior of the Kafka client used with Quarkus 3.8 has changed in comparison to Quarkus 3.2. I have tried to adapt accordingly but my understanding of our KafkaConsumer is limited. Would you mind taking a look what is going on?

**Comment by calohmn** [comment]
@sophokles73 Ok, I'll check.

**Comment by calohmn** [comment]
@sophokles73 Tests were failing with Kafka 3.6.1 because of changes to the `MockConsumer.rebalance()` method (see [here](https://github.com/apache/kafka/compare/3.4.0...3.6.1#diff-0ebcab2d2727abe3f0f1e3a94321b068ef2478911a33861d072a7a2ecbb1cf1f)).
Previously, the `onPartitionsRevoked` and `onPartitionsAssigned` handlers were not invoked there and we had used our own implementation in the Hono `KafkaMockConsumer` subclass.
The Hono implementation was mimicking the bahviour of the _eager_ rebalance protocol, with the existing partitions being revoked first and then the new assignment getting applied.
The new `MockConsumer.rebalance()` implementation isn't doing that and is therefore mimicking the _cooperative_ rebalance protocol.
Adding support for the old behaviour would require the use of reflection.
But since we are using cooperative rebalancing by default anyway (see https://github.com/eclipse-hono/hono/pull/2908), I think it's better to concentrate on that.
I've adapted the tests accordingly.

File: bom/pom.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
     <cryptvault.version>1.0.2</cryptvault.version>
     <dispatch-router.image.name>quay.io/interconnectedcloud/qdrouterd:1.17.1</dispatch-router.image.name>
     <guava.version>32.1.2-jre</guava.version>
-    <infinispan.version>14.0.11.Final</infinispan.version>
+    <infinispan.version>14.0.27.Final</infinispan.version>
```
Comment (calohmn) [inline bom/pom.xml:37]: `DEPENDENCIES` file should be regenerated for this.

File: services/command-router-base/pom.xml [hunk 2]
```diff
@@ -102,6 +102,18 @@
...
+      <artifactId>okhttp</artifactId>
+      <version>3.14.9</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.squareup.okio</groupId>
+      <artifactId>okio</artifactId>
+      <version>1.17.6</version>
+      <scope>test</scope>
+    </dependency>
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply services/command-router-base/pom.xml]: Thanks for the hint, I have pushed a fix.

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**